azurelinuxagent/common/utils/networkutil.py

```diff
@@ -200,21 +248,27 @@ def __execute_cmd(cmd):
 
     @staticmethod
     def add_iptables_rules(wait, dst_ip, uid):
-
+        # The order in which the below rules are added matters for the ip table rules to work as expected
         AddFirewallRules.__raise_if_empty(dst_ip, "Destination IP")
         AddFirewallRules.__raise_if_empty(uid, "User ID")
 
-        accept_rule = AddFirewallRules.get_iptables_accept_command(wait, AddFirewallRules.__ACCEPT_COMMAND, dst_ip, uid)
+        accept_rule_dns_tcp_rule = AddFirewallRules.get_iptables_accept_dns_tcp_request_command(wait, AddFirewallRules.get_insert_command(),dst_ip)
+        AddFirewallRules.__execute_cmd(accept_rule_dns_tcp_rule)
+
+        accept_rule = AddFirewallRules.get_iptables_accept_command(wait, AddFirewallRules.get_append_command(), dst_ip, uid)
         AddFirewallRules.__execute_cmd(accept_rule)
 
-        drop_rule = AddFirewallRules.get_iptables_drop_command(wait, AddFirewallRules.__ACCEPT_COMMAND, dst_ip)
+        drop_rule = AddFirewallRules.get_iptables_drop_command(wait, AddFirewallRules.get_append_command(), dst_ip)
```

Comment:
These names were a bit confusion, could you rename all of them to `get_iptables_drop_rule` instead of `get_iptables_drop_command`. Similarly for the others too (`get_iptables_accept_rule` and `get_iptables_accept_dns_tcp_request_rule`)

Reply:
I have modified the name as ""get_firewall_accept_dns_tcp_rule" consistently. For next PR after I change the implementation of other two rules I will modified the names of the to use "rule" everywhere